.github/workflows/tests.yaml

```diff
@@ -0,0 +1,110 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '**'
+
+env:
+  BEE_VERSION: '1.5.1-d0a77598-stateful'
+  BLOCKCHAIN_VERSION: '1.2.0-for-1.5.1-d0a77598-stateful'
+  BEE_ENV_PREFIX: 'swarm-test'
+  BEE_IMAGE_PREFIX: 'docker.pkg.github.com/ethersphere/bee-factory'
+  COMMIT_VERSION_TAG: 'false'
+  WORKERS: 4
+  BEE_API_URL: 'http://127.0.0.1:1633'
+  BEE_PEER_API_URL: 'http://127.0.0.1:11633'
+  BEE_DEBUG_API_URL: 'http://127.0.0.1:1635'
+  BEE_PEER_DEBUG_API_URL: 'http://127.0.0.1:11635'
+  BEE_TEST_CHEQUEBOOK: true
+
+jobs:
+  nodejs:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Auth to Github Package Docker Registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login https://docker.pkg.github.com -u ${GITHUB_ACTOR} --password-stdin
```

Comment:
it seems right for builds made by `ethersphere`, but we will have our own build by `fairDataSociety/bee-factory`